src/app/tools/symmetry.h

```diff
@@ -9,24 +10,50 @@
 #pragma once
 
 #include "app/tools/stroke.h"
+#include "app/pref/preferences.h"
 
-#include <vector>
+#include <memory>
 
 namespace app {
-  namespace tools {
-
-    class ToolLoop;
-
-    // This class controls user input.
-    class Symmetry {
-    public:
-      virtual ~Symmetry() { }
-
-      // The "stroke" must be relative to the sprite origin.
-      virtual void generateStrokes(const Stroke& stroke, Strokes& strokes, ToolLoop* loop) = 0;
-    };
-
-  } // namespace tools
+namespace tools {
+
+class ToolLoop;
+
+class Symmetry {
+public:
+  Symmetry(gen::SymmetryMode symmetryMode, double x, double y) : m_symmetryMode(symmetryMode)
+  {
+    switch (symmetryMode) {
+      case gen::SymmetryMode::HORIZONTAL:
+        m_x = x;
+      case gen::SymmetryMode::VERTICAL:
+        m_y = y;
+        break;
+      case gen::SymmetryMode::BOTH:
+        m_x = x;
+        m_y = y;
+        break;
+      default:
+        ASSERT(false)
+        m_symmetryMode = gen::SymmetryMode::NONE;
+        break;
+    }
+  }
```

Comment:
Simplify constructor:
```suggestion
  Symmetry(gen::SymmetryMode symmetryMode, double x, double y)
    : m_symmetryMode(symmetryMode)
    , m_x(x), m_y(y) {
  }
```